### ISSUE TITLE
Bolt's bundled modules pollute the module namespace with common names

### ISSUE THREAD

**Issue by lollipopman** [puppetlabs/bolt#3055]
## Describe the Bug

```
bolt apply manifest.pp
Failed on butter.example.com:
  Apply failed to compile for butter.example.com: Unknown resource type: 'system::role' (file: manifest.pp, line: 5, column: 5)
```

## Expected Behavior

Bolt would find the system module, or print a loud warning that it was shadowed by a built in bolt
module.

## Additional Context

I found it very surprising that Bolt was polluting the global module namespace with very common module names, could it instead use modules prefixed with bolt or puppetlabs perhaps?

```
boltlib/
ctrl/
dir/
file/
log/
out/
prompt/
system/
```

**Comment by donoghuc** [comment]
repro case:

The following fails:
```
(base) ➜  bolt git:(adde8ca7) ✗ cat modules/system/manifests/foo.pp
class system::foo{
	notify{foo:}
}%
(base) ➜  bolt git:(adde8ca7) ✗ bundle exec bolt apply -e "include system::foo" -t localhost --format json --modulepath modules
[{"target":"localhost","action":"apply","object":null,"status":"failure","value":{"_error":{"kind":"bolt/apply-error","msg":"Apply failed to compile for localhost: Could not find class ::system::foo for localhost (line: 1, column: 1)","details":{}}}}]
````

Without a module naming conflict the class would have been found:
```
(base) ➜  bolt git:(adde8ca7) ✗ cat modules/foo/manifests/bar.pp
class foo::bar{
	notify{foo:}
}%
(base) ➜  bolt git:(adde8ca7) ✗ bundle exec bolt apply -e "include foo::bar" -t localhost --format json --modulepath modules
[{"target":"localhost","action":"apply","object":null,"status":"success","value":{"report":{"host":"cass-macbook-pro.local","time":"2022-02-14T13:43:11.005264000-08:00","configuration_version":1644874987,"transaction_uuid":null,"report_format":12,"puppet_version":"7.13.1","status":"changed","transaction_completed":true,"noop":false,"noop_pending":false,"environment":null,"logs":[{"level":"notice","message":"foo","source":"Puppet","tags":["notice"],"time":"2022-02-14T13:43:11.032817000-08:00","file":null,"line":null},{"level":"notice","message":"defined 'message' as 'foo'","source":"/Stage[main]/Foo::Bar/Notify[foo]/message","tags":["notice","notify","foo","class","foo::bar","bar"],"time":"2022-02-14T13:43:11.033305000-08:00","file":"/Users/cas.donoghue/puppetlabs/bolt/modules/foo/manifests/bar.pp","line":2},{"level":"notice","message":"Applied catalog in 0.00 seconds","source":"Puppet","tags":["notice"],"time":"2022-02-14T13:43:11.035677000-08:00","file":null,"line":null}],"metrics":{"resources":{"name":"resources","label":"Resources","values":[["total","Total",1],["skipped","Skipped",0],["failed","Failed",0],["failed_to_restart","Failed to restart",0],["restarted","Restarted",0],["changed","Changed",1],["out_of_sync","Out of sync",1],["scheduled","Scheduled",0],["corrective_change","Corrective change",0]]},"time":{"name":"time","label":"Time","values":[["notify","Notify",0.000696],["config_retrieval","Config retrieval",0],["transaction_evaluation","Transaction evaluation",0.0027269998099654913],["catalog_application","Catalog application",0.004830999998375773],["total","Total",0.030436]]},"changes":{"name":"changes","label":"Changes","values":[["total","Total",1]]},"events":{"name":"events","label":"Events","values":[["total","Total",1],["failure","Failure",0],["success","Success",1]]}},"resource_statuses":{"Notify[foo]":{"title":"foo","file":"/Users/cas.donoghue/puppetlabs/bolt/modules/foo/manifests/bar.pp","line":2,"resource":"Notify[foo]","resource_type":"Notify","provider_used":null,"containment_path":["Stage[main]","Foo::Bar","Notify[foo]"],"evaluation_time":0.000696,"tags":["notify","foo","class","foo::bar","bar"],"time":"2022-02-14T13:43:11.032662000-08:00","failed":false,"failed_to_restart":false,"changed":true,"out_of_sync":true,"skipped":false,"change_count":1,"out_of_sync_count":1,"events":[{"audited":false,"property":"message","previous_value":"absent","desired_value":"foo","historical_value":null,"message":"defined 'message' as 'foo'","name":"message_changed","status":"success","time":"2022-02-14T13:43:11.032773000-08:00","redacted":null,"corrective_change":false}],"corrective_change":false}},"corrective_change":false,"catalog_uuid":"6e3fd4d3-b8ef-42f4-b522-aeb1b369fca8","cached_catalog_status":"not_used"},"_output":"changed: 1, failed: 0, unchanged: 0 skipped: 0, noop: 0"}}]
```

**Comment by lollipopman** [comment]
@donoghuc thanks, should have included a minimal example in my original report!

**Comment by donoghuc** [comment]
I have not had time to think about how to solve this. I see how it is annoying, but just prefixing it with `bolt` would be a very unattractive breaking change. One potential option could be to move these out of `bolt-lib` to modules somehow, then if your module shadows the core content, its on the author to deal with it. But again, i've not really thought through the consequences. Maybe I can spend some time on this in the coming weeks.

**Comment by lollipopman** [comment]
@donoghuc I agree it is a difficult decision. I looked around for general advice for handling avoiding puppet module naming collisions, but I did not find anything too helpful. Though prefixing bolt's modules with `bolt` would be a painful breaking change, I still see that as a better option than the status quo, at least for modules like `system` which have very generic names. In contrast with a module name such as `nginx` where duplicate names are more expected, but they are usually less an issue since you probably only need to use one nginx module in your codebase. Putting the onus on the user of bolt to change their module names, seems a bit unfriendly. For instance our `system` module in our repo dates to 2013, which is four years before the bolt project made its first commit!
